src/Sylius/Bundle/ApiBundle/Resources/config/routing/product.yml

```diff
@@ -21,23 +21,23 @@ sylius_api_product_create:
             form: sylius_api_product
 
 sylius_api_product_update:
-    path: /{id}
+    path: /{code}
     methods: [POST, PUT, PATCH]
     defaults:
         _controller: sylius.controller.product:updateAction
         _sylius:
             form: sylius_api_product
```

Comment:
@lchrusciel deleting the `from` declaration is the way we're currently going ;).